index.html

```diff
@@ -569,70 +634,95 @@
     var doInit = function() {
       var s = $location.search();
       $scope.map = s.map || "US";
-      $scope.orig = (s.orig === undefined) ? -1 : +s.orig;
-      $scope.dest = (s.dest === undefined) ? -1 : +s.dest;
-      $scope.playerColor = (s.color === undefined) ? "white" : s.color;
+      try {
+        $scope.players = JSON.parse(s.players);
+      } catch (e) {
+        $scope.players = [];
+      }
+      // There should always be at least one player
+      $scope.numPlayers = Math.max(1, $scope.players.length);
+      $scope.updateNumPlayers();
     };
-    $scope.$on("$locationChangeSuccess", doInit);
     doInit();
   });
+
+  /**
+   * Allow representing models directly as numbers, not strings.
+   */
+  app.directive('convertToNumber', function() {
```

Comment:
Nope -- I originally had it set up on the numPlayers select box, but that got refactored away. Deleted.